File: src/server/router/schedule.go
package router

import (
	"yule-go/handler"
	"yule-go/repository"
	"yule-go/service"

	"github.com/gin-gonic/gin"
)

// RegisterScheduleRoutes 注册团期相关路由
func RegisterScheduleRoutes(r *gin.RouterGroup) {
	// 初始化依赖链
	repo := repository.NewScheduleRepository()
	svc := service.NewScheduleService(repo)
	h := handler.NewScheduleHandler(svc)

	// 管理后台路由（需要管理员认证）
	admin := r.Group("/admin")
	{
		admin.POST("/schedules", h.Create)
		admin.PUT("/schedules/:id", h.Update)
		admin.PUT("/schedules/:id/cancel", h.Cancel)
		admin.GET("/schedules/:id", h.GetByID)
		admin.GET("/schedules", h.List)
	}

	// 小程序端路由（需要 JWT 认证）
	public := r.Group("")
	{
		public.GET("/schedules", h.ListByWeek)
	}
}

File: src/web/src/api/schedule.js
import api from './index'

export const scheduleApi = {
  // 创建团期
  create(data) {
    return api.post('/admin/schedules', data)
  },

  // 编辑团期
  update(id, data) {
    return api.put(`/admin/schedules/${id}`, data)
  },

  // 取消团期
  cancel(id) {
    return api.put(`/admin/schedules/${id}/cancel`)
  },

  // 团期详情
  getById(id) {
    return api.get(`/admin/schedules/${id}`)
  },

  // 团期列表（分页）
  list(params) {
    return api.get('/admin/schedules', { params })
  },

  // 按周查询（小程序端）
  listByWeek(week) {
    return api.get('/schedules', { params: { week } })
  }
}
